tests/aclorch_ut.cpp

```diff
@@ -0,0 +1,1075 @@
+extern MirrorOrch* gMirrorOrch;
+extern VRFOrch* gVrfOrch;
+
+extern sai_acl_api_t* sai_acl_api;
+extern sai_switch_api_t* sai_switch_api;
+extern sai_port_api_t* sai_port_api;
+extern sai_vlan_api_t* sai_vlan_api;
+extern sai_bridge_api_t* sai_bridge_api;
+extern sai_route_api_t* sai_route_api;
+
+namespace nsAclOrchTest {
+
+using namespace std;
+
+static size_t consumerAddToSync(Consumer* consumer, const std::deque<KeyOpFieldsValuesTuple>& entries)
+{
+    /* Nothing popped */
+    if (entries.empty()) {
+        return 0;
+    }
+
+    for (auto& entry : entries) {
+        string key = kfvKey(entry);
+        string op = kfvOp(entry);
+
+        /* If a new task comes or if a DEL task comes, we directly put it into getConsumerTable().m_toSync map */
+        if (consumer->m_toSync.find(key) == consumer->m_toSync.end() || op == DEL_COMMAND) {
+            consumer->m_toSync[key] = entry;
+        }
+        /* If an old task is still there, we combine the old task with new task */
+        else {
+            KeyOpFieldsValuesTuple existing_data = consumer->m_toSync[key];
+
+            auto new_values = kfvFieldsValues(entry);
+            auto existing_values = kfvFieldsValues(existing_data);
+
+            for (auto it : new_values) {
+                string field = fvField(it);
+                string value = fvValue(it);
+
+                auto iu = existing_values.begin();
+                while (iu != existing_values.end()) {
+                    string ofield = fvField(*iu);
+                    if (field == ofield)
+                        iu = existing_values.erase(iu);
+                    else
+                        iu++;
+                }
+                existing_values.push_back(FieldValueTuple(field, value));
+            }
+            consumer->m_toSync[key] = KeyOpFieldsValuesTuple(key, op, existing_values);
+        }
+    }
+    return entries.size();
+}
+
+struct AclTestBase : public ::testing::Test {
+    std::vector<int32_t*> m_s32list_pool;
+
+    virtual ~AclTestBase()
+    {
+        for (auto p : m_s32list_pool) {
+            free(p);
+        }
+    }
+};
+
+struct AclTest : public AclTestBase {
+
+    struct CreateAclResult {
+        bool ret_val;
+
+        std::vector<sai_attribute_t> attr_list;
+    };
+
+    std::shared_ptr<swss::DBConnector> m_config_db;
+
+    AclTest()
+    {
+        m_config_db = std::make_shared<swss::DBConnector>(CONFIG_DB, swss::DBConnector::DEFAULT_UNIXSOCKET, 0);
+    }
+
+    void SetUp() override
+    {
+        assert(gCrmOrch == nullptr);
+        gCrmOrch = new CrmOrch(m_config_db.get(), CFG_CRM_TABLE_NAME);
+    }
+
+    void TearDown() override
+    {
+        delete gCrmOrch;
+        gCrmOrch = nullptr;
+    }
+
+    std::shared_ptr<CreateAclResult> createAclTable(AclTable& acl)
+    {
+        assert(sai_acl_api == nullptr);
+
+        sai_acl_api = new sai_acl_api_t();
+        auto sai_acl = std::shared_ptr<sai_acl_api_t>(sai_acl_api, [](sai_acl_api_t* p) {
```

Comment:
>std::shared_ptr<sai_acl_api_t> [](start = 23, length = 30)

use make_shared() directly?

Reply:
No. Because I want to reset `sai_acl_api`. `make_shared()` under C++11, can not pass a customized remove function.